dbms/src/Flash/Pipeline/Schedule/Events/FineGrainedPipelineEvent.cpp

```diff
@@ -0,0 +1,27 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Flash/Pipeline/Schedule/Events/FineGrainedPipelineEvent.h>
+#include <Flash/Pipeline/Schedule/Tasks/PipelineTask.h>
+
+namespace DB
+{
+std::vector<TaskPtr> FineGrainedPipelineEvent::scheduleImpl()
+{
+    std::vector<TaskPtr> tasks;
+    tasks.reserve(1);
```

Comment:
For just one element vector, reserve seems not efficient.

Reply:
ok, removed.